packages/insomnia/src/ui/routes/runner.tsx

```
@@ -37,57 +36,93 @@ const inputStyle = 'placeholder:italic py-0.5 mr-1.5 px-1 w-24 rounded-sm border
 
 // TODO: improve the performance for a lot of logs
-async function aggregateAllTimelines(testResult: RunnerTestResult) {
-  const responsesInfo = testResult.responsesInfo;
+async function aggregateAllTimelines(errorMsg: string | null, testResult: RunnerTestResult) {
   let timelines = new Array<ResponseTimelineEntry>();
 
-  for (let i = 0; i < responsesInfo.length; i++) {
-    const respInfo = responsesInfo[i];
-    const resp = await models.response.getById(respInfo.responseId);
-    if (resp) {
-      const timeline = models.response.getTimeline(resp, true) as unknown as ResponseTimelineEntry[];
-      timelines = [
-        ...timelines,
-        {
-          value: `------ Start of request (${respInfo.originalRequestName}) ------\n\n\n`,
-          name: 'Text',
-          timestamp: Date.now(),
-        },
-        ...timeline,
-      ];
-    } else {
-      console.error(`failed to read response for the request ${respInfo.originalRequestName}`);
+  if (errorMsg) {
+    timelines = [
+      {
+        value: errorMsg,
+        name: 'Text',
+        timestamp: Date.now(),
+      },
+    ];
+  } else {
+    const responsesInfo = testResult.responsesInfo;
+
+    for (let i = 0; i < responsesInfo.length; i++) {
+      const respInfo = responsesInfo[i];
+      const resp = await models.response.getById(respInfo.responseId);
+
+      if (resp) {
+        const timeline = models.response.getTimeline(resp, true) as unknown as ResponseTimelineEntry[];
+        timelines = [
+          ...timelines,
+          {
+            value: `------ Start of request (${respInfo.originalRequestName}) ------\n\n\n`,
+            name: 'Text',
+            timestamp: Date.now(),
+          },
+          ...timeline,
+        ];
+      } else {
+        console.error(`failed to read response for the request ${respInfo.originalRequestName}`);
+      }
```

I think it would be better here to show something in the timeline that the response failed to be read instead of logging to console. So we don't have to look in two places.

Also since its only a finite number of elements it would be easier to read and debug, and more consistent with the codebase if it was written as a map reduce style rather than a forloop.

---

Good idea and logged error to the console. The forloop is because of an async operation inside it.

---

since the async operation is happening for every item in the list you can write a simple map and assign it to a variable first.

---

I tried several ways and mapping to promises may require some logic handling errors and ordering. Will revisit it when I find a solution in the repo.